src/main/java/fi/hsl/transitdata/tripupdate/models/PubtransData.java

```diff
@@ -0,0 +1,56 @@
+package fi.hsl.transitdata.tripupdate.models;
+
+import fi.hsl.common.transitdata.proto.InternalMessages;
+import fi.hsl.common.transitdata.proto.PubtransTableProtos;
+import fi.hsl.transitdata.tripupdate.processing.BaseProcessor;
+
+/**
+ * Intermediate data format which will be refactored out soon, we'll just read StopEstimates directly.
+ */
+public class PubtransData {
+
+    public PubtransTableProtos.Common common;
+    public PubtransTableProtos.DOITripInfo tripInfo;
+    public BaseProcessor.EventType eventType;
+
+    public PubtransData(BaseProcessor.EventType eventType, PubtransTableProtos.Common common, PubtransTableProtos.DOITripInfo tripInfo) {
+        this.tripInfo = tripInfo;
+        this.common = common;
+        this.eventType = eventType;
+    }
+
+    /**
+     * StopEstimate is the data-model that we're going to use in the future so let's refactor this codebase already to use it internally.
+     * Then it's easy to swap to new input schema.
+     *
+     * Throws RuntimeException if data is not valid.
+     */
+    public InternalMessages.StopEstimate toStopEstimate() throws Exception {
+        InternalMessages.StopEstimate.Builder builder = InternalMessages.StopEstimate.newBuilder();
+        builder.setSchemaVersion(builder.getSchemaVersion());
+
+        InternalMessages.TripInfo.Builder tripBuilder = InternalMessages.TripInfo.newBuilder();
+        tripBuilder.setTripId(Long.toString(tripInfo.getDvjId()));
+        tripBuilder.setOperatingDay(tripInfo.getOperatingDay());
+        tripBuilder.setRouteId(tripInfo.getRouteId());
+        tripBuilder.setDirectionId(tripInfo.getDirectionId());//Jore format
+        tripBuilder.setStartTime(tripInfo.getStartTime());
+        builder.setTripInfo(tripBuilder.build());
+        builder.setStopId(tripInfo.getStopId());
+        builder.setStopSequence(common.getJourneyPatternSequenceNumber());
+
+        InternalMessages.StopEstimate.Status scheduledStatus = (common.getState() == 3L) ?
```

Comment:
Is there any good way of making the `common.getState()` (and other proto enums) actual enums?

Reply:
Proto enums are found as java enums also, but this particular case (common.getState()) is a direct value from a database. We could of course convert that to string/enum in the SQL but the reason for 3L was the original: we wanted to keep the raw-data as-is. This demand however is slipping away so we could convert the code to more readable format. 

also the parsing goes away to StopEstimates-microservice so TripUpdate actually can work on better abstractions.